Update EIP-5805: Refer to EIP-6372 for clock and move to Review (#6484)
ethereum/EIPs: EIPS/eip-5805.md

### Methods

#### [ERC-6372](./eip-6372.md): clock and CLOCK_MODE

> A contract implementing this ERC may optionally implement ERC-6372 (Contract clock) to announce the clock that is used for vote tracking.

Reviewer: ```suggestion
A contract implementing this ERC SHOULD implement ERC-6372 (Contract clock) to announce the clock that is used for vote tracking.
```

Author: The rest of the ERC isn't using MUST, SHOULD, etc.

If we want to use those we need a larger review. Let's do that in another PR.

Author: Oh, my bad I see that it is using those keywords. Just need to add the notice about them.

Reviewer 2: Yeah, basically you can choose to adopt it or not. I just wanna point out it leaves ambiguity in some cases and could cause challenges for adoption (the contract client side)

### Methods

> #### [ERC-6372](./eip-6372.md): clock and CLOCK_MODE

Reviewer 3: ```suggestion
#### [ERC-6372](./eip-6372.md): clock and `CLOCK_MODE`
```
